docs/deployment.rst

```diff
@@ -0,0 +1,66 @@
+Configuration and deployment
+############################
+
+codejail-service has some unusual requirements due to its sensitive nature. **Please read all sections carefully.**
+
+Before you begin
+****************
+
+Ensure your intended deployment environment meets the following prerequisites:
+
+* Ubuntu, Debian, or another Linux distribution that supports AppArmor
+* AppArmor 3 or newer (e.g. in Ubuntu 22.04 or newer)
+* The ability to load an AppArmor profile into the host and (if using Docker) apply the profile to a Docker container
+
+Network
+*******
+
+If at all possible, your codejail-service cluster should be run with the following networking restrictions:
+
+* No inbound access from the public internet. codejail-service lacks authentication, rate-limiting, audit logs, and other protections, and its functionality should only be exposed via Custom Python-evaluated Input Problems in the LMS and CMS.
+* No outbound connections, *especially* to other services within your deployment. This will help limit the impact if an attacker were to figure out how to escape the sandbox.
+
+Host
+****
+
+The host or Docker image must be set up with a sandbox user, sandbox virtualenv, and sudoers file according to the documentation of the `codejail library <https://github.com/openedx/codejail>`__. This forms the foundation of the sandboxing, but does not provide much security by itself.
+
+See `2U's Dockerfile <https://github.com/edx/public-dockerfiles/blob/main/dockerfiles/codejail-service.Dockerfile>`__ for an example of how you could set up your host or container. (There is not yet a Tutor plugin.)
+
+AppArmor
+********
+
+Again following the codejail library's instructions, create an AppArmor profile and load it into the host.
+
+This profile will need to be customized according to your service's app user, sandbox setup, and other details. It will also look different if you are using Docker or running the service dierctly on the host. For reference, here is `2U's AppArmor profile <https://github.com/edx/public-dockerfiles/blob/main/apparmor/openedx_codejail_service.profile>`__; note that the inner profile is the one that actually applies the sandboxing.
```

Comment:
```suggestion
This profile will need to be customized according to your service's app user, sandbox setup, and other details. It will also look different if you are using Docker or running the service directly on the host. For reference, here is `2U's AppArmor profile <https://github.com/edx/public-dockerfiles/blob/main/apparmor/openedx_codejail_service.profile>`__; note that the inner profile is the one that actually applies the sandboxing.
```